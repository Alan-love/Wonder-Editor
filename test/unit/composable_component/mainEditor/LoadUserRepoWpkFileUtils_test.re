open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

open Header;

open Js.Promise;

let _ =
  describe("LoadUserRepoWpkFileUtils", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => {
      sandbox := createSandbox();

      ImportPackageTool.prepareLoad(sandbox);

      MainEditorSceneTool.initState(~sandbox, ());

      MainEditorSceneTool.createDefaultSceneAndNotInit(sandbox);

      MainEditorAssetTreeTool.BuildAssetTree.buildEmptyAssetTree() |> ignore;
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("load and import user repo wpk file from server", () => {
      beforeEach(() =>
        UserDataTool.setUserData |> StateLogicService.getAndSetEditorState
      );

      testPromise(
        {|
            export fake wpk w1;
            load and import w1;

            sceneTree should has w1->gameObjects;
            |},
        () => {
          MainEditorLeftHeaderTool.addCube();
          let wpkArrayBuffer = ExportPackageTool.exportWPK();

          let fetch =
            BuildFetchTool.buildFakeFetch(
              BuildFetchTool.buildFakeFetchResponse(wpkArrayBuffer),
            );
          let editorState = StateEditorService.getState();
          let dispatchFuncStub = ReactTool.createDispatchFuncStub(sandbox);

          LoadUserRepoWpkFileUtils.loadUserRepoWpkFile(
            dispatchFuncStub,
            fetch,
            editorState,
          )
          |> WonderBsMost.Most.drain
          |> Js.Promise.then_(_ =>
               BuildComponentTool.buildSceneTree(
                 TestTool.buildEmptyAppState(),
               )
               |> ReactTestTool.createSnapshotAndMatch
               |> resolve
             );
        },
      );
    });
  });