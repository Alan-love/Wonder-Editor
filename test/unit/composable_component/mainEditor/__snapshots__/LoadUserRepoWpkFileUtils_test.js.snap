// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadUserRepoWpkFileUtils load and import user repo wpk file from server 
            export fake wpk w1;
            load and import w1;

            sceneTree should has w1->gameObjects;
             1`] = `
<article
  className="wonder-sceneTree-component"
  id="wonder-sceneTree-component"
>
  <article
    className="wonder-tree"
  >
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
          onMouseDown={[Function]}
        >
          <img
            src="./public/img/down.png"
          />
        </div>
        <div
          className="draggable-container  no-drag"
          draggable={false}
          id="sceneTreeNode-2"
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          <img
            className="treeNode-icon"
            src="./public/img/gameObjectIcon.png"
          />
          Scene
        </div>
      </li>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-3"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            <img
              className="treeNode-icon"
              src="./public/img/gameObjectIcon.png"
            />
            Cube
          </div>
        </li>
      </ul>
    </ul>
  </article>
</article>
`;
